test/src/compose/application-manager.spec.ts

```diff
@@ -900,19 +931,80 @@ describe('compose/application-manager', () => {
 		expect(nextSteps).to.have.lengthOf(0);
 	});
 
-	it('should infer that an image should be removed if it is no longer referenced in current or target state', async () => {
+	it('should infer that an image should be removed if it is no longer referenced in current or target state (only target)', async () => {
 		const targetApps = createApps(
 			{
 				services: [
 					await createService(
 						{ image: 'main-image' },
+						// Target has a matching image already
 						{ options: { imageInfo: { Id: 'sha256:bbbb' } } },
 					),
 				],
 				networks: [DEFAULT_NETWORK],
 			},
 			true,
 		);
+		const {
+			currentApps,
+			availableImages,
+			downloading,
+			containerIdsByAppId,
+		} = createCurrentState({
+			services: [],
+			networks: [DEFAULT_NETWORK],
+			images: [
+				// An image for a service that no longer exists
+				{
+					name: 'old-image',
+					appId: 5,
+					serviceId: 5,
+					serviceName: 'old-service',
+					imageId: 5,
+					dependent: 0,
+					releaseId: 5,
+					dockerImageId: 'sha256:aaaa',
+				},
+				{
+					name: 'main-image',
+					appId: 1,
+					serviceId: 1,
+					serviceName: 'main',
+					imageId: 1,
+					dependent: 0,
+					releaseId: 1,
+					dockerImageId: 'sha256:bbbb',
+				},
+			],
+		});
+
+		const [removeImageStep] = await applicationManager.inferNextSteps(
+			currentApps,
+			targetApps,
+			{
+				downloading,
+				availableImages,
+				containerIdsByAppId,
+			},
+		);
+
+		// A start step shoud happen for the depended service first
```

Comment:
```suggestion
		// A start step should happen for the depended service first
```

Reply:
@pipex you marked as resolved but the comment still has the type

Reply:
No it doesn't. Not sure why it shows like this, but on the final file it has the correct spelling